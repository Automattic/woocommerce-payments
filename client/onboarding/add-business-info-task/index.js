/**
 * External dependencies
 */
import React, { useState } from 'react';
//  import { useEffect, useContext } from '@wordpress/element';
import { __ } from '@wordpress/i18n';
import { SelectControl } from '@wordpress/components';
// import { useDispatch } from '@wordpress/data';

/**
 * Internal dependencies
 */
import WizardTaskItem from 'wcpay/additional-methods-setup/wizard/task-item';

const AddBusinessInfo = () => {
	const [ country, businessType, setCountry, setBusinessType ] = useState();

	return (
		<WizardTaskItem
			className="complete-business-info-task"
			index={ 1 }
			title={ __(
				'Tell us more about your business',
				'woocommerce-payments'
			) }
			visibleDescription={ __(
				"Preview the details we'll require to verify your business and enable deposits.",
				'woocommerce-payments'
			) }
		>
			<SelectControl
				label={ __( 'Country', 'woocommerce-payments' ) }
				value={ country }
				onChange={ ( value ) => setCountry( value ) }
				options={ [
					{ label: 'United States', value: 'US' },
					{
						label: 'United Kingdom',
						value: 'UK',
					},
				] }
			/>

			<p>
				{ __(
					'The primary country where your business operates',
					'woocommerce-payments'
				) }
			</p>

			<SelectControl
				label={ __( 'Business type', 'woocommerce-payments' ) }
				value={ businessType }
				onChange={ ( value ) => setBusinessType( value ) }
				options={ [
					{
						label: 'Individual',
						value: 'individual',
					},
					{ label: 'Company', value: 'company' },
					{
						label: 'Non-profit Organisation',
						value: 'nonprofit_organisation',
					},
				] }
			/>
		</WizardTaskItem>
	);
};

export default AddBusinessInfo;
