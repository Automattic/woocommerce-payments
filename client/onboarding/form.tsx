/**
 * External dependencies
 */
import React from 'react';
import { Button } from '@wordpress/components';
import { isEmpty, mapValues } from 'lodash';

/**
 * Internal dependencies
 */
import { useStepperContext } from 'components/stepper';
import { Item as SelectItem } from 'components/custom-select-control';
import { ListItem as GroupedSelectItem } from 'components/grouped-select-control';
import {
	GroupedSelectField,
	GroupedSelectFieldProps,
	SelectField,
	SelectFieldProps,
	TextField,
	TextFieldProps,
} from 'components/form/fields';
import { useOnboardingContext } from './context';
import { OnboardingFields } from './types';
import { useValidation } from './validation';
import { trackStepCompleted } from './tracking';
import strings from './strings';

export const OnboardingForm: React.FC = ( { children } ) => {
	const { errors, touched, setTouched } = useOnboardingContext();
	const { currentStep, nextStep } = useStepperContext();

	const handleContinue = () => {
		if ( isEmpty( errors ) ) {
			trackStepCompleted( currentStep );
			return nextStep();
		}
		setTouched( mapValues( touched, () => true ) );
	};

	return (
		<form
			onSubmit={ ( event ) => {
				event.preventDefault();
				handleContinue();
			} }
		>
			{ children }
			<Button
				variant={ 'primary' }
				type="submit"
				className="stepper__cta"
			>
				{ strings.continue }
			</Button>
		</form>
	);
};

interface OnboardingTextFieldProps extends Partial< TextFieldProps > {
	name: keyof OnboardingFields;
}

export const OnboardingTextField: React.FC< OnboardingTextFieldProps > = (
	props
) => {
	const { name } = props;
	const { data, setData, touched } = useOnboardingContext();
	const { validate, error } = useValidation( name );
	const inputRef = React.useRef< HTMLInputElement >( null );

	return (
		<TextField
			ref={ inputRef as any }
			label={ strings.fields[ name ] }
			value={ data[ name ] || '' }
			onChange={ ( value: string ) => {
				setData( { [ name ]: value } );
				if (
					touched[ name ] ||
					inputRef.current !==
						inputRef.current?.ownerDocument.activeElement
				)
					validate( value );
			} }
			onBlur={ () => validate() }
			onKeyDown={ ( event: React.KeyboardEvent< HTMLInputElement > ) => {
				if ( event.key === 'Enter' ) validate();
			} }
			error={ error() }
			{ ...props }
		/>
	);
};

interface OnboardingSelectFieldProps< ItemType >
	extends Partial< Omit< SelectFieldProps< ItemType >, 'onChange' > > {
	name: keyof OnboardingFields;
	onChange?: ( name: keyof OnboardingFields, item?: ItemType | null ) => void;
}

export const OnboardingSelectField = < ItemType extends SelectItem >( {
	onChange,
	...rest
}: OnboardingSelectFieldProps< ItemType > ): JSX.Element => {
	const { name } = rest;
	const { data, setData } = useOnboardingContext();
	const { validate, error } = useValidation( name );

	return (
		<SelectField
			label={ strings.fields[ name ] }
			value={ rest.options?.find(
				( item ) => item.key === data[ name ]
			) }
			placeholder={
				( strings.placeholders as Record< string, string > )[ name ] ??
				strings.placeholders.generic
			}
			onChange={ ( { selectedItem } ) => {
				if ( onChange ) {
					onChange?.( name, selectedItem );
				} else {
					setData( { [ name ]: selectedItem?.key } );
				}
				validate( selectedItem?.key );
			} }
			options={ [] }
			error={ error() }
			{ ...rest }
		/>
	);
};

interface OnboardingGroupedSelectFieldProps< ItemType >
	extends Partial< Omit< GroupedSelectFieldProps< ItemType >, 'onChange' > > {
	name: keyof OnboardingFields;
	onChange?: ( name: keyof OnboardingFields, item?: ItemType | null ) => void;
}

export const OnboardingGroupedSelectField = <
	ListItemType extends GroupedSelectItem
>( {
	onChange,
	...rest
}: OnboardingGroupedSelectFieldProps< ListItemType > ): JSX.Element => {
	const { name } = rest;
	const { data, setData } = useOnboardingContext();
	const { validate, error } = useValidation( name );

	return (
		<GroupedSelectField
			label={ strings.fields[ name ] }
			value={ rest.options?.find(
				( item ) => item.key === data[ name ]
			) }
			placeholder={
				( strings.placeholders as Record< string, string > )[ name ] ??
				strings.placeholders.generic
			}
			onChange={ ( { selectedItem } ) => {
				if ( onChange ) {
					onChange?.( name, selectedItem );
				} else {
					setData( { [ name ]: selectedItem?.key } );
				}
				validate( selectedItem?.key );
			} }
			options={ [] }
			error={ error() }
			{ ...rest }
		/>
	);
};
