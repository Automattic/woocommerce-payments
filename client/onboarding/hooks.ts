/**
 * External dependencies
 */
import { addQueryArgs } from '@wordpress/url';
import { __ } from '@wordpress/i18n';
import { NAMESPACE } from 'data/constants';

/**
 * Internal dependencies
 */
import { useApiFetch } from 'hooks/use-api-fetch';
import { AccountSession, OnboardingProps } from './types';

type AccountSessionData = AccountSession;

interface AccountSessionReturn {
	accountSession: AccountSession;
	isLoading: boolean;
}

export const useAccountSession = (
	query: OnboardingProps
): AccountSessionReturn => {
	const path = addQueryArgs( `${ NAMESPACE }/onboarding/embedded`, query );
	const { data, isLoading } = useApiFetch< AccountSessionData >( {
		path,
		errorMessage: __(
			'Error creating account session',
			'woocommerce-payments'
		),
	} );

	return {
		accountSession: {
			clientSecret: data?.clientSecret || '',
			expiresAt: data?.expiresAt || 0,
			accountId: data?.accountId || '',
			isLive: data?.isLive || false,
			accountCreated: data?.accountCreated || false,
		},
		isLoading,
	};
};
