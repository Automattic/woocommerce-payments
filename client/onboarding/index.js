/** @format */

/**
 * External dependencies
 */
import React from 'react';
import { Card, CardBody } from '@wordpress/components';

/**
 * Internal dependencies
 */
import Wizard from 'additional-methods-setup/wizard/wrapper';
import WizardTaskList from 'additional-methods-setup/wizard/task-list';
import WizardTask from 'additional-methods-setup/wizard/task';
import AddBusinessInfo from './tasks/add-business-info-task';
import SetupComplete from './tasks/setup-complete-task';

import './index.scss';

const OnboardingPage = () => {
	return (
		<Card
			size="large"
			className="wcpay-onboarding-card wcpay-homescreen-card"
		>
			<CardBody>
				<Wizard defaultActiveTask="complete-business-info">
					<WizardTaskList>
						<WizardTask id="complete-business-info">
							<AddBusinessInfo />
						</WizardTask>
						<WizardTask id="setup-complete">
							<SetupComplete />
						</WizardTask>
					</WizardTaskList>
				</Wizard>
			</CardBody>
		</Card>
	);
};

export default OnboardingPage;
