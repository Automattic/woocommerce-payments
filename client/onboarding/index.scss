.wcpay-wizard-task__description-element.subheading {
	font-size: 0.9rem;
}

.wcpay-onboarding-select-business-types {
	margin-bottom: 8px;

	& button.components-custom-select-control__button {
		width: 100%;
		font-size: 17px;

		@media ( min-width: 600px ) {
			font-size: 14px;
		}
	}
}

.wcpay-onboarding-setup-complete-task__buttons {
	margin-top: 1rem;
}
