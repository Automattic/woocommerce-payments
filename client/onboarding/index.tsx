/**
 * External dependencies
 */
import React, { useState } from 'react';
import { Card, CardBody } from '@wordpress/components';

/**
 * Internal dependencies
 */
import Wizard from 'additional-methods-setup/wizard/wrapper';
import WizardTaskList from 'additional-methods-setup/wizard/task-list';
import WizardTask from 'additional-methods-setup/wizard/task';
import AddBusinessInfoTask from './tasks/add-business-info-task';
import SetupCompleteTask from './tasks/setup-complete-task';
import { OnboardingProps } from './types';
import './style.scss';

const OnboardingPage = (): JSX.Element => {
	const [ state, setState ] = useState< Partial< OnboardingProps > >( {} );

	return (
		<Card
			size="large"
			className="wcpay-onboarding-card wcpay-homescreen-card"
		>
			<CardBody>
				<Wizard defaultActiveTask="complete-business-info">
					<WizardTaskList>
						<WizardTask id="complete-business-info">
							<AddBusinessInfoTask onChange={ setState } />
						</WizardTask>
						<WizardTask id="setup-complete">
							<SetupCompleteTask args={ state } />
						</WizardTask>
					</WizardTaskList>
				</Wizard>
			</CardBody>
		</Card>
	);
};

export default OnboardingPage;
