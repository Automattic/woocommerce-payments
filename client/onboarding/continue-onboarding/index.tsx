/**
 * External dependencies
 */
import React, { useEffect } from 'react';

/**
 * Internal dependencies
 */
import Page from 'wcpay/components/page';
import { OnboardingContextProvider } from 'wcpay/onboarding/context';
import EmbeddedOnboarding from 'wcpay/onboarding/steps/embedded-onboarding';
import Logo from 'assets/images/woopayments.svg';
import { closeSmall, Icon } from '@wordpress/icons';
import strings from 'wcpay/onboarding/strings';
import { getAdminUrl } from 'wcpay/utils';

const ContinueOnboardingPage: React.FC = () => {
	const handleExit = () => {
		const urlParams = new URLSearchParams( window.location.search );

		window.location.href = getAdminUrl( {
			page: 'wc-admin',
			path: '/payments/connect',
			source:
				urlParams.get( 'source' )?.replace( /[^\w-]+/g, '' ) ||
				'unknown',
			from: 'WCPAY_ONBOARDING_WIZARD',
		} );
	};

	useEffect( () => {
		// Remove loading class and add those required for full screen.
		document.body.classList.remove( 'woocommerce-admin-is-loading' );
		document.body.classList.add( 'woocommerce-admin-full-screen' );
		document.body.classList.add( 'is-wp-toolbar-disabled' );
		document.body.classList.add( 'wcpay-onboarding__body' );

		// Remove full screen classes on unmount.
		return () => {
			document.body.classList.remove( 'woocommerce-admin-full-screen' );
			document.body.classList.remove( 'is-wp-toolbar-disabled' );
			document.body.classList.remove( 'wcpay-onboarding__body' );
		};
	}, [] );
	return (
		<Page className="wcpay-onboarding-prototype">
			<OnboardingContextProvider>
				<div className="stepper__nav">
					<button
						type="button"
						className={ `stepper__nav-button hide` }
					>
						{ strings.back }
					</button>
					<img
						src={ Logo }
						alt="WooPayments"
						className="stepper__nav-logo"
					/>
					<button
						type="button"
						className="stepper__nav-button"
						onClick={ handleExit }
					>
						<Icon icon={ closeSmall } />
					</button>
				</div>
				<div className="stepper__wrapper">
					<div className="stepper__content">
						<EmbeddedOnboarding />
					</div>
				</div>
			</OnboardingContextProvider>
		</Page>
	);
};
export default ContinueOnboardingPage;
