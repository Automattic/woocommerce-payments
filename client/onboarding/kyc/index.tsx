/**
 * External dependencies
 */
import React, { useEffect } from 'react';

/**
 * Internal dependencies
 */
import WooLogo from 'assets/images/woo-logo.svg';
import Page from 'components/page';
import { OnboardingContextProvider } from 'onboarding/context';
import EmbeddedKyc from 'onboarding/steps/embedded-kyc';
import strings from 'onboarding/strings';
import { getConnectUrl } from 'utils';
import { trackKycExit } from 'wcpay/onboarding/tracking';

const OnboardingKycPage: React.FC = () => {
	const urlParams = new URLSearchParams( window.location.search );
	const collectPayoutRequirements = !! urlParams.get(
		'collect_payout_requirements'
	);

	const handleExit = () => {
		trackKycExit();

		// Let the connect logic determine where the merchant should end up.
		window.location.href = getConnectUrl(
			{
				source:
					urlParams.get( 'source' )?.replace( /[^\w-]+/g, '' ) ||
					'unknown',
			},
			'WCPAY_ONBOARDING_KYC'
		);
	};

	useEffect( () => {
		// Remove loading class and add those required for full screen.
		document.body.classList.remove( 'woocommerce-admin-is-loading' );
		document.body.classList.add( 'woocommerce-admin-full-screen' );
		document.body.classList.add( 'is-wp-toolbar-disabled' );
		document.body.classList.add( 'wcpay-onboarding__body' );

		// Remove full screen classes on unmount.
		return () => {
			document.body.classList.remove( 'woocommerce-admin-full-screen' );
			document.body.classList.remove( 'is-wp-toolbar-disabled' );
			document.body.classList.remove( 'wcpay-onboarding__body' );
		};
	}, [] );
	return (
		<Page className="wcpay-onboarding-prototype">
			<OnboardingContextProvider>
				<div className="stepper__nav">
					<button
						type="button"
						className={ `stepper__nav-button hide` }
					>
						{ strings.back }
					</button>
					<img
						src={ WooLogo }
						alt="Woo"
						className="stepper__nav-logo"
					/>
					<button
						type="button"
						className="stepper__nav-button"
						onClick={ handleExit }
					>
						{ strings.cancel }
					</button>
				</div>
				<div className="stepper__wrapper">
					<div className="stepper__content">
						<EmbeddedKyc
							continueKyc={ true }
							collectPayoutRequirements={
								collectPayoutRequirements
							}
						/>
					</div>
				</div>
			</OnboardingContextProvider>
		</Page>
	);
};
export default OnboardingKycPage;
