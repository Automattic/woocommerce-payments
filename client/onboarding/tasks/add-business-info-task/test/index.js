/** @format */

/**
 * External dependencies
 */
import { render, screen } from '@testing-library/react';
import user from '@testing-library/user-event';
import React from 'react';

/**
 * Internal dependencies
 */
import AddBusinessInfoTask from '../';
import { useBusinessTypes } from 'wcpay/data/onboarding';

jest.mock( 'wcpay/data/onboarding', () => ( {
	useBusinessTypes: jest.fn(),
} ) );

const businessTypes = [
	{
		key: 'US',
		name: 'United States',
		types: [
			{
				key: 'individual',
				name: 'Individual',
				structures: [],
			},
			{
				key: 'company',
				name: 'Company',
				structures: [
					{
						key: 'sole_proprietorship',
						name: 'Sole propietorship',
					},
					{
						key: 'single_member_llc',
						name: 'Single member LLC',
					},
				],
			},
		],
	},
	{
		key: 'FR',
		name: 'France',
		types: [
			{
				key: 'individual',
				name: 'Individual',
				structures: [],
			},
			{
				key: 'company',
				name: 'Company',
				structures: [],
			},
			{
				key: 'non_profit',
				name: 'Non-profit',
				structures: [],
			},
		],
	},
];

const renderTask = () =>
	render( <AddBusinessInfoTask onChange={ () => {} } /> );

describe( 'AddBusinessInfoTask', () => {
	beforeEach( () => {
		jest.clearAllMocks();
		global.wcpaySettings = {
			connect: {
				country: 'US',
			},
		};
	} );

	it( 'shows a loadable block', () => {
		useBusinessTypes.mockReturnValue( {
			businessTypes: [],
			isLoading: true,
		} );

		const { container: task } = renderTask();
		expect( task ).toMatchSnapshot();
	} );

	it( 'shows the form', () => {
		useBusinessTypes.mockReturnValue( {
			businessTypes,
			isLoading: false,
		} );

		const { container: task } = renderTask();
		expect( task ).toMatchSnapshot();
	} );

	it( 'shows business type and structure from a selected country', () => {
		global.wcpaySettings = {
			connect: {
				country: 'FR',
			},
		};
		useBusinessTypes.mockReturnValue( {
			businessTypes,
			isLoading: false,
		} );

		const { container: task } = renderTask();

		user.click( screen.getByRole( 'button', { name: /country/i } ) );
		user.click(
			screen.getByRole( 'option', {
				name: /united states/i,
			} )
		);

		user.click( screen.getByRole( 'button', { name: /business type/i } ) );
		user.click(
			screen.getByRole( 'option', {
				name: /company/i,
			} )
		);

		user.click(
			screen.getByRole( 'button', { name: /business structure/i } )
		);
		user.click(
			screen.getByRole( 'option', {
				name: /single member llc/i,
			} )
		);

		expect( task ).toMatchSnapshot();
	} );

	it( 'hides the structure when no structures are available', () => {
		global.wcpaySettings = {
			connect: {
				country: 'FR',
			},
		};
		useBusinessTypes.mockReturnValue( {
			businessTypes,
			isLoading: false,
		} );

		const { container: task } = renderTask();

		user.click( screen.getByRole( 'button', { name: /country/i } ) );
		user.click(
			screen.getByRole( 'option', {
				name: /united states/i,
			} )
		);

		user.click( screen.getByRole( 'button', { name: /business type/i } ) );
		user.click(
			screen.getByRole( 'option', {
				name: /individual/i,
			} )
		);

		expect( task ).toMatchSnapshot();
		expect(
			screen.queryByRole( 'button', { name: /business structure/i } )
		).not.toBeInTheDocument();
	} );
} );
