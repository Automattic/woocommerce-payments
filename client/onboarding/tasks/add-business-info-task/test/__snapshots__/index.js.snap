// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddBusinessInfoTask hides the structure when no structures are available 1`] = `
<div>
  <li
    class="wcpay-wizard-task complete-business-info-task"
  >
    <div
      class="wcpay-wizard-task__top-border"
    />
    <div
      class="wcpay-wizard-task__headline"
      tabindex="-1"
    >
      <div
        class="wcpay-wizard-task__icon-wrapper"
      >
        <div
          class="wcpay-wizard-task__icon-text"
        >
          1
        </div>
        <svg
          aria-hidden="true"
          class="wcpay-wizard-task__icon-checkmark"
          focusable="false"
          height="24"
          role="img"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M18.3 5.6L9.9 16.9l-4.6-3.4-.9 1.2 5.8 4.3 9.3-12.6z"
          />
        </svg>
      </div>
      <span
        class="wcpay-wizard-task__title"
      >
        Tell us more about your business
      </span>
    </div>
    <div
      class="wcpay-wizard-task__body"
    >
      <p
        class="complete-business-info-task__subheading"
      >
        Preview the details we may require to verify your business and enable despoits.
      </p>
      <div
        class="wcpay components-custom-select-control"
      >
        <label
          class="components-custom-select-control__label"
          for="downshift-23-toggle-button"
          id="downshift-23-label"
        >
          Country
        </label>
        <button
          aria-describedby="Currently selected: United States"
          aria-expanded="false"
          aria-haspopup="listbox"
          aria-label="Country"
          class="components-button components-custom-select-control__button"
          id="downshift-23-toggle-button"
          type="button"
        >
          United States
          <svg
            aria-hidden="true"
            class="components-custom-select-control__button-icon"
            focusable="false"
            height="24"
            role="img"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M17.5 11.6L12 16l-5.5-4.4.9-1.2L12 14l4.5-3.6 1 1.2z"
            />
          </svg>
        </button>
        <ul
          aria-hidden="true"
          aria-labelledby="downshift-23-label"
          class="components-custom-select-control__menu"
          id="downshift-23-menu"
          role="listbox"
          tabindex="-1"
        />
      </div>
      <p
        class="complete-business-info-task__description"
      >
        The primary country where your business operates
      </p>
      <div
        class="wcpay components-custom-select-control"
      >
        <label
          class="components-custom-select-control__label"
          for="downshift-25-toggle-button"
          id="downshift-25-label"
        >
          Business type
        </label>
        <button
          aria-describedby="Currently selected: Individual"
          aria-expanded="false"
          aria-haspopup="listbox"
          aria-label="Business type"
          class="components-button components-custom-select-control__button"
          id="downshift-25-toggle-button"
          type="button"
        >
          Individual
          <svg
            aria-hidden="true"
            class="components-custom-select-control__button-icon"
            focusable="false"
            height="24"
            role="img"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M17.5 11.6L12 16l-5.5-4.4.9-1.2L12 14l4.5-3.6 1 1.2z"
            />
          </svg>
        </button>
        <ul
          aria-hidden="true"
          aria-labelledby="downshift-25-label"
          class="components-custom-select-control__menu"
          id="downshift-25-menu"
          role="listbox"
          tabindex="-1"
        />
      </div>
    </div>
  </li>
</div>
`;

exports[`AddBusinessInfoTask shows a loadable block 1`] = `
<div>
  <li
    class="wcpay-wizard-task complete-business-info-task"
  >
    <div
      class="wcpay-wizard-task__top-border"
    />
    <div
      class="wcpay-wizard-task__headline"
      tabindex="-1"
    >
      <div
        class="wcpay-wizard-task__icon-wrapper"
      >
        <div
          class="wcpay-wizard-task__icon-text"
        >
          1
        </div>
        <svg
          aria-hidden="true"
          class="wcpay-wizard-task__icon-checkmark"
          focusable="false"
          height="24"
          role="img"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M18.3 5.6L9.9 16.9l-4.6-3.4-.9 1.2 5.8 4.3 9.3-12.6z"
          />
        </svg>
      </div>
      <span
        class="wcpay-wizard-task__title"
      >
        Tell us more about your business
      </span>
    </div>
    <div
      class="wcpay-wizard-task__body"
    >
      <p
        class="complete-business-info-task__subheading"
      >
        Preview the details we may require to verify your business and enable despoits.
      </p>
      <span
        aria-busy="true"
        class="is-loadable-placeholder is-block"
      >
        <p
          style="line-height: 4;"
        >
          Block placeholder
        </p>
      </span>
      <span
        aria-busy="true"
        class="is-loadable-placeholder is-block"
      >
        <p
          style="line-height: 4;"
        >
          Block placeholder
        </p>
      </span>
    </div>
  </li>
</div>
`;

exports[`AddBusinessInfoTask shows business type and structure from a selected country 1`] = `
<div>
  <li
    class="wcpay-wizard-task complete-business-info-task"
  >
    <div
      class="wcpay-wizard-task__top-border"
    />
    <div
      class="wcpay-wizard-task__headline"
      tabindex="-1"
    >
      <div
        class="wcpay-wizard-task__icon-wrapper"
      >
        <div
          class="wcpay-wizard-task__icon-text"
        >
          1
        </div>
        <svg
          aria-hidden="true"
          class="wcpay-wizard-task__icon-checkmark"
          focusable="false"
          height="24"
          role="img"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M18.3 5.6L9.9 16.9l-4.6-3.4-.9 1.2 5.8 4.3 9.3-12.6z"
          />
        </svg>
      </div>
      <span
        class="wcpay-wizard-task__title"
      >
        Tell us more about your business
      </span>
    </div>
    <div
      class="wcpay-wizard-task__body"
    >
      <p
        class="complete-business-info-task__subheading"
      >
        Preview the details we may require to verify your business and enable despoits.
      </p>
      <div
        class="wcpay components-custom-select-control"
      >
        <label
          class="components-custom-select-control__label"
          for="downshift-3-toggle-button"
          id="downshift-3-label"
        >
          Country
        </label>
        <button
          aria-describedby="Currently selected: United States"
          aria-expanded="false"
          aria-haspopup="listbox"
          aria-label="Country"
          class="components-button components-custom-select-control__button"
          id="downshift-3-toggle-button"
          type="button"
        >
          United States
          <svg
            aria-hidden="true"
            class="components-custom-select-control__button-icon"
            focusable="false"
            height="24"
            role="img"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M17.5 11.6L12 16l-5.5-4.4.9-1.2L12 14l4.5-3.6 1 1.2z"
            />
          </svg>
        </button>
        <ul
          aria-hidden="true"
          aria-labelledby="downshift-3-label"
          class="components-custom-select-control__menu"
          id="downshift-3-menu"
          role="listbox"
          tabindex="-1"
        />
      </div>
      <p
        class="complete-business-info-task__description"
      >
        The primary country where your business operates
      </p>
      <div
        class="wcpay components-custom-select-control"
      >
        <label
          class="components-custom-select-control__label"
          for="downshift-5-toggle-button"
          id="downshift-5-label"
        >
          Business type
        </label>
        <button
          aria-describedby="Currently selected: Company"
          aria-expanded="false"
          aria-haspopup="listbox"
          aria-label="Business type"
          class="components-button components-custom-select-control__button"
          id="downshift-5-toggle-button"
          type="button"
        >
          Company
          <svg
            aria-hidden="true"
            class="components-custom-select-control__button-icon"
            focusable="false"
            height="24"
            role="img"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M17.5 11.6L12 16l-5.5-4.4.9-1.2L12 14l4.5-3.6 1 1.2z"
            />
          </svg>
        </button>
        <ul
          aria-hidden="true"
          aria-labelledby="downshift-5-label"
          class="components-custom-select-control__menu"
          id="downshift-5-menu"
          role="listbox"
          tabindex="-1"
        />
      </div>
      <div
        class="wcpay components-custom-select-control"
      >
        <label
          class="components-custom-select-control__label"
          for="downshift-16-toggle-button"
          id="downshift-16-label"
        >
          Business Structure
        </label>
        <button
          aria-describedby="Currently selected: Single member LLC"
          aria-expanded="false"
          aria-haspopup="listbox"
          aria-label="Business Structure"
          class="components-button components-custom-select-control__button"
          id="downshift-16-toggle-button"
          type="button"
        >
          Single member LLC
          <svg
            aria-hidden="true"
            class="components-custom-select-control__button-icon"
            focusable="false"
            height="24"
            role="img"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M17.5 11.6L12 16l-5.5-4.4.9-1.2L12 14l4.5-3.6 1 1.2z"
            />
          </svg>
        </button>
        <ul
          aria-hidden="true"
          aria-labelledby="downshift-16-label"
          class="components-custom-select-control__menu"
          id="downshift-16-menu"
          role="listbox"
          tabindex="-1"
        />
      </div>
    </div>
  </li>
</div>
`;

exports[`AddBusinessInfoTask shows the form 1`] = `
<div>
  <li
    class="wcpay-wizard-task complete-business-info-task"
  >
    <div
      class="wcpay-wizard-task__top-border"
    />
    <div
      class="wcpay-wizard-task__headline"
      tabindex="-1"
    >
      <div
        class="wcpay-wizard-task__icon-wrapper"
      >
        <div
          class="wcpay-wizard-task__icon-text"
        >
          1
        </div>
        <svg
          aria-hidden="true"
          class="wcpay-wizard-task__icon-checkmark"
          focusable="false"
          height="24"
          role="img"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M18.3 5.6L9.9 16.9l-4.6-3.4-.9 1.2 5.8 4.3 9.3-12.6z"
          />
        </svg>
      </div>
      <span
        class="wcpay-wizard-task__title"
      >
        Tell us more about your business
      </span>
    </div>
    <div
      class="wcpay-wizard-task__body"
    >
      <p
        class="complete-business-info-task__subheading"
      >
        Preview the details we may require to verify your business and enable despoits.
      </p>
      <div
        class="wcpay components-custom-select-control"
      >
        <label
          class="components-custom-select-control__label"
          for="downshift-0-toggle-button"
          id="downshift-0-label"
        >
          Country
        </label>
        <button
          aria-describedby="Currently selected: United States"
          aria-expanded="false"
          aria-haspopup="listbox"
          aria-label="Country"
          class="components-button components-custom-select-control__button"
          id="downshift-0-toggle-button"
          type="button"
        >
          United States
          <svg
            aria-hidden="true"
            class="components-custom-select-control__button-icon"
            focusable="false"
            height="24"
            role="img"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M17.5 11.6L12 16l-5.5-4.4.9-1.2L12 14l4.5-3.6 1 1.2z"
            />
          </svg>
        </button>
        <ul
          aria-hidden="true"
          aria-labelledby="downshift-0-label"
          class="components-custom-select-control__menu"
          id="downshift-0-menu"
          role="listbox"
          tabindex="-1"
        />
      </div>
      <p
        class="complete-business-info-task__description"
      >
        The primary country where your business operates
      </p>
      <div
        class="wcpay components-custom-select-control"
      >
        <label
          class="components-custom-select-control__label"
          for="downshift-2-toggle-button"
          id="downshift-2-label"
        >
          Business type
        </label>
        <button
          aria-describedby="No selection"
          aria-expanded="false"
          aria-haspopup="listbox"
          aria-label="Business type"
          class="components-button components-custom-select-control__button placeholder"
          id="downshift-2-toggle-button"
          type="button"
        >
          What type of business do you run?
          <svg
            aria-hidden="true"
            class="components-custom-select-control__button-icon"
            focusable="false"
            height="24"
            role="img"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M17.5 11.6L12 16l-5.5-4.4.9-1.2L12 14l4.5-3.6 1 1.2z"
            />
          </svg>
        </button>
        <ul
          aria-hidden="true"
          aria-labelledby="downshift-2-label"
          class="components-custom-select-control__menu"
          id="downshift-2-menu"
          role="listbox"
          tabindex="-1"
        />
      </div>
      
    </div>
  </li>
</div>
`;
